test/types/reply.test-d.ts

```diff
@@ -14,8 +14,9 @@ const getHandler: RouteHandlerMethod = function (_request, reply) {
   expectType<FastifyContextConfig>(reply.context.config)
   expectType<FastifyLoggerInstance>(reply.log)
   expectType<FastifyRequest<RouteGenericInterface, RawServerDefault, RawRequestDefaultExpression>>(reply.request)
-  expectType<(statusCode: number) => FastifyReply>(reply.code)
-  expectType<(statusCode: number) => FastifyReply>(reply.status)
+  type ModifiedReplyType<Code extends number> = FastifyReply<RawServerDefault, RawRequestDefaultExpression, RawReplyDefaultExpression, RouteGenericInterface, ContextConfigDefault, FastifySchema, FastifyTypeProviderDefault, ReplyTypeInfer<Code>, Code>
```

Comment:
We need a test to see if `return reply.code(200).send({})` still possible.

Reply:
Would something like this `expectType<(payload?: unknown) => ModifiedReplyType>(reply.status(200).send)` be satisfactory?

Reply:
I would assume the below code do not throw error.
Of cuz, the `server` should be using `type-provider`.

```ts
server.get('/allow-narrowed-type-return', {
  schema: {
    response: {
      200: { type: 'string' }
    }
  }
}, function handler (request, reply) {
  return reply.code(200).send('abc')
})

server.get('/allow-narrowed-type-return', {
  schema: {
    response: {
      200: { type: 'string' }
    }
  }
}, async function handler (request, reply) {
  return reply.code(200).send('abc')
})
```

Reply:
No, this should not throw an error.

Reply:
I just want a test to ensure it works.

Reply:
Sure, I will add it.